src/renderer/modals/SignTransaction/Body.js

```diff
@@ -0,0 +1,248 @@
+    footer: StepConfirmationFooter,
+    onBack: ({ transitionTo, onRetry }) => {
+      onRetry();
+      transitionTo("amount");
+    },
+  },
+];
+
+const STATUS_KEYS_IGNORE = ["recipient", "gasLimit"];
+
+// returns the first error
+function getStatusError(status, type = "errors"): ?Error {
+  if (!status || !status[type]) return null;
+
+  const firstKey = Object.keys(status[type]).filter(k => !STATUS_KEYS_IGNORE.includes(k))[0];
+
+  return firstKey ? status[type][firstKey] : null;
+}
+
+const mapStateToProps = createStructuredSelector({
+  device: getCurrentDevice,
+  accounts: accountsSelector,
+});
+
+const mapDispatchToProps = {
+  closeModal,
+  openModal,
+  updateAccountWithUpdater,
+};
+
+const Body = ({
+  t,
+  device,
+  openModal,
+  closeModal,
+  onChangeStepId,
+  onClose,
+  setError,
+  stepId,
+  params,
+  accounts,
+}: Props) => {
+  const openedFromAccount = !!params.account;
+  const [steps] = useState(createSteps);
+
+  const {
+    transaction,
+    setTransaction,
+    updateTransaction,
+    account,
+    parentAccount,
+    setAccount,
+    status,
+    bridgeError,
+    bridgePending,
+  } = useBridgeTransaction(() => {
+    const parentAccount = params && params.parentAccount;
+    const account = getMainAccount((params && params.account) || accounts[0], parentAccount);
+
+    const bridge = getAccountBridge(account, parentAccount);
+    const t = bridge.createTransaction(account);
+    const { recipient, ...txData } = params.transactionData;
+    const t2 = bridge.updateTransaction(t, {
+      recipient,
+      feesStrategy: "custom",
+    });
+    const transaction = bridge.updateTransaction(t2, {
+      ...txData,
+      userGasLimit: txData.gasLimit,
+    });
+
+    return { account, parentAccount, transaction };
+  });
+
+  const [transactionError, setTransactionError] = useState(null);
+
+  const handleCloseModal = useCallback(() => {
+    closeModal("MODAL_SIGN_TRANSACTION");
+  }, [closeModal]);
+
+  const handleChangeAccount = useCallback(
+    (nextAccount: AccountLike, nextParentAccount: ?Account) => {
+      if (account !== nextAccount) {
+        setAccount(nextAccount, nextParentAccount);
+      }
+    },
+    [account, setAccount],
+  );
+
+  const handleRetry = useCallback(() => {
+    setTransactionError(null);
+    setError(undefined);
+  }, [setError]);
+
+  const handleTransactionError = useCallback(
+    (error: Error) => {
+      if (!(error instanceof UserRefusedOnDevice)) {
+        logger.critical(error);
+      }
+      setTransactionError(error);
+      setError(error);
+    },
+    [setError],
+  );
+
+  const handleStepChange = useCallback(e => onChangeStepId(e.id), [onChangeStepId]);
+
+  const handleTransactionSigned = useCallback(
+    (signedTransaction: SignedOperation) => {
+      params.onResult(signedTransaction);
+      handleCloseModal();
+    },
+    [handleCloseModal, params],
+  );
+
+  const errorSteps = [];
+
+  if (transactionError) {
+    errorSteps.push(3);
```

Comment:
what does 3 means?

Reply:
i think we can remove it here its supposed to be the index of the stepper but since theres only 2 steps no need to show on the 3rd :laughing: